src/components/input/index.tsx

```diff
@@ -45,9 +53,58 @@ export type InputProps = {
  *
  * @description is a component that is used to get user input in a text field.
  */
-export const Input = forwardRef<InputProps, 'input'>((props, ref) => {
-  const { label, hint, error, disabled, leftIcon, rightIcon, css, ...rest } =
-    props;
+export const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
+  const [hasValue, setHasValue] = useState(false);
+  const innerRef = useRef<HTMLInputElement | null>();
+
+  const {
+    label,
+    hint,
+    error,
+    disabled,
+    leftIcon,
+    rightIcon,
+    cleanable,
+    css,
+    onChange,
+    ...rest
+  } = props;
+
+  if (!!rightIcon && cleanable) {
+    throw new Error("You can't use both 'rightIcon' and 'cleanable' props");
+  }
+
+  const setMultipleRefs = useCallback(
+    (element: HTMLInputElement) => {
+      innerRef.current = element;
+
+      if (typeof ref === 'function') {
+        ref(element);
+      }
+
+      if (ref && typeof ref !== 'function') {
+        ref.current = element;
+      }
+    },
+    [ref, innerRef],
+  );
+
```

Comment:
```suggestion
  const setMultipleRefs = useCallback(
    (element: HTMLInputElement) => {
      innerRef.current = element;

      if (typeof ref === 'function') {
        ref(element);
      }

      if (ref && typeof ref !== 'function') {
        ref.current = element;
      }
    },
    [ref, innerRef],
  );
```
This can be abstracted to a system hook later, I think that we will use this more this